server/schedulers/hot_region.go

```diff
@@ -633,22 +489,41 @@ func (bs *balanceSolver) tryAddPendingInfluence() bool {
 	switch {
 	case bs.rwTy == write && bs.opTy == transferLeader:
 		maxZombieDur = bs.sche.conf.GetRegionsStatZombieDuration()
+	case bs.rwTy == write && bs.opTy == movePeer:
+		if bs.best.srcDetail.Info.IsTiFlash || bs.best.dstDetail.Info.IsTiFlash {
+			maxZombieDur = bs.sche.conf.GetRegionsStatZombieDuration()
+		} else {
+			maxZombieDur = bs.sche.conf.GetStoreStatZombieDuration()
+		}
 	default:
 		maxZombieDur = bs.sche.conf.GetStoreStatZombieDuration()
 	}
-	return bs.sche.tryAddPendingInfluence(bs.ops[0], bs.best.srcStoreID, bs.best.dstStoreID, bs.infl, maxZombieDur)
+	return bs.sche.tryAddPendingInfluence(bs.ops[0], bs.best.srcDetail.Info.Store.GetID(), bs.best.dstDetail.Info.Store.GetID(), bs.infl, maxZombieDur)
 }
 
 // filterSrcStores compare the min rate and the ratio * expectation rate, if two dim rate is greater than
 // its expectation * ratio, the store would be selected as hot source store
 func (bs *balanceSolver) filterSrcStores() map[uint64]*storeLoadDetail {
 	ret := make(map[uint64]*storeLoadDetail)
+	confSrcToleranceRatio := bs.sche.conf.GetSrcToleranceRatio()
+	confEnableForTiFlash := bs.sche.conf.GetEnableForTiFlash()
 	for id, detail := range bs.stLoadDetail {
+		srcToleranceRatio := confSrcToleranceRatio
+		if detail.Info.IsTiFlash {
```

Comment:
```suggestion
		if detail.Info.IsTiFlash  &&  confEnableForTiFlash{
```

Reply:
> Is there any unit test?

After dealing with the problem of the code.